### PR TITLE
Add cmd/docgen for document generator

### LINKED ISSUE #1032: Add cmd/docgen for document generator
Part of #1028.

1. The executable `docgen` will update the documentation according to the latest attribute dictionary.
1. Our pre-commit hooks and continuous integration should run the executable `docgen`.

### PR DESCRIPTION
**Pull request by tonyyang-svail**
Fix #1032.

### CODE REVIEW

File: doc/model.md [hunk 1]
```diff
@@ -1,10 +1,12 @@
-# XGBoost Parameter Document
+# Model Document
```
Comment (wangkuiyi) [inline doc/model.md]: What is a "Model Document"? The title is very confusing. I can understand a file named `pkg/sql/codegen/xgboost/README.md` is a about the XGBoost codegen; however, I have no idea what will a `model.md` file is supposed to present.

File: scripts/docgen.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+set -e
+
+go run cmd/docgen/main.go > doc/model.md
```
Comment (wangkuiyi) [inline scripts/docgen.sh]: If the executable is supposed to update `doc/model.md`, why don't name it `docgen/model.go`?

**Review by wangkuiyi — COMMENTED** [review]
I can understand the purpose of this PR from the issue, but I feel the implementation is somehow confusing.

File: doc/model.md (hunk 1 above)
Comment (tonyyang-svail) [reply doc/model.md]: How about renaming it to "Model Parameter Document"? As mentioned in the first paragraph of this document, the document lists all model parameters supported by SQLFlow

File: scripts/docgen.sh (hunk 2 above)
Comment (tonyyang-svail) [reply scripts/docgen.sh]: Sure. I have renamed `docgen/main.go` to `docgen/model_parameter.go`.

File: doc/model.md (hunk 1 above)
Comment (tonyyang-svail) [reply doc/model.md]: You are right, the name `model.md` is confusing. I will change it to `model_parameter.md`.

File: cmd/docgen/model_parameter.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"sqlflow.org/sqlflow/pkg/sql/codegen/xgboost"
+)
+
+func main() {
+	fmt.Print(`# Model Parameter Document
```
Comment (typhoonzero) [inline cmd/docgen/model_parameter.go:25]: `sqlflow.org` will generate indexes using document titles. Should have meaningful titles so the document for tensorflow, xgboost, alps wouldn't be the same.

Comment (tonyyang-svail) [reply cmd/docgen/model_parameter.go:25]: Sorry about the confusion. I want this file contains all the document for for tensorflow, xgboost, alps.

I have updated the description in https://github.com/sql-machine-learning/sqlflow/issues/1028 to the following.

> A command-line `docgen` in `cmd/docgen` that will call `func DocGenInMarkdown() string` for each package and combine them to `doc/model_parameters.md`

Comment (typhoonzero) [reply cmd/docgen/model_parameter.go:25]: I see

**Review by typhoonzero — APPROVED** [review]
LGTM

**Pull request merged**